server/container/guice/mailet/src/main/java/org/apache/james/utils/GuiceMailetLoader.java

```diff
@@ -43,17 +43,22 @@ public class GuiceMailetLoader implements MailetLoader {
     public GuiceMailetLoader(GuiceGenericLoader genericLoader, Set<MailetConfigurationOverride> mailetConfigurationOverrides) {
         this.genericLoader = genericLoader;
         this.configurationOverrides = mailetConfigurationOverrides.stream()
-            .collect(ImmutableMap.toImmutableMap(
-                MailetConfigurationOverride::getClazz,
-                MailetConfigurationOverride::getNewConfiguration));
+                .collect(ImmutableMap.toImmutableMap(
+                        MailetConfigurationOverride::getClazz,
+                        MailetConfigurationOverride::getNewConfiguration));
     }
 
     @Override
     public Mailet getMailet(MailetConfig config) throws MessagingException {
         try {
             ClassName className = new ClassName(config.getMailetName());
-            Mailet result = genericLoader.<Mailet>withNamingSheme(MAILET_NAMING_SCHEME)
-                .instantiate(className);
+            var mailetLoader = genericLoader.<Mailet>withNamingSheme(MAILET_NAMING_SCHEME);
+            Class<Mailet> mailetClass = mailetLoader.locateClass(className);
+            MailetConfig mailetConfig = resolveConfiguration(mailetClass, config);
+            Mailet result = mailetLoader.withChildModule(
+                            binder -> binder.bind(MailetConfig.class).toInstance(mailetConfig)
+                    )
```

Comment:
```suggestion
            Mailet result = mailetLoader.withChildModule(
                            binder -> binder.bind(MailetConfig.class).toInstance(mailetConfig))
```

(sorry)

Reply:
I fixed the indentation let me know if it's ok for you

Reply:
My point is more that I am opinionated on lines starting with brackets but that may not matter much.